entra_id/credentials_provider.go

```diff
@@ -0,0 +1,57 @@
+package entra_id
+
+import (
+	"log"
+	"time"
+)
+
+// EntraIdIdentityProvider defines the interface for an identity provider
+type EntraIdIdentityProvider interface {
```

Comment:
If we think about how this interface will look when called outside of this package it will be `entra_id.EntraIdIdentityProvidor` .  As you can see there is a repetition - `EntraId` which we drop. Here is a brief section related to repetition in the [go styleguide](https://google.github.io/styleguide/go/decisions#repetition)